operator/src/main/java/org/bf2/operator/events/ControllerEventFilter.java

```diff
@@ -0,0 +1,84 @@
+package org.bf2.operator.events;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import io.fabric8.zjsonpatch.JsonDiff;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+import io.javaoperatorsdk.operator.processing.event.source.controller.ResourceEventFilter;
+import org.jboss.logging.Logger;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Resource event filter for use with
+ * {@link io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration#eventFilters
+ * eventFilters} used by {@code ManagedKafka} and {@code ManagedKafkaAgent}
+ * controllers. This event filter will accept any change where the resource:
+ * <ul>
+ * <li>is added
+ * <li>generation has changed
+ * <li>annotations have changed
+ * <li>labels have changed
+ * </ul>
+ *
+ */
+public class ControllerEventFilter implements ResourceEventFilter<HasMetadata> {
+
+    private static Logger log = Logger.getLogger(ControllerEventFilter.class);
+
+    @Override
+    public boolean acceptChange(ControllerConfiguration<HasMetadata> configuration,
+            HasMetadata oldResource,
+            HasMetadata newResource) {
+
+        Optional<ObjectMeta> oldMeta = meta(oldResource);
+        Optional<ObjectMeta> newMeta = meta(newResource);
+
+        boolean resourceChanged =
+                oldResource == null || // Always accept an "add" change
+                changed(oldMeta, newMeta, this::generation) ||
```

Comment:
I'll make a note to have fabric8 expose some logic for this - it's highly related to how it produces diffs by removing kubernetes managed fields.  More generally there's a bunch of other fields that could change, like the ownerReference, that we don't care about here but could be something built-in for the operator sdk - "generationAndMetaAware" or something.